projects/RabbitMQ.Client/client/impl/MethodArgumentReader.cs

```diff
@@ -81,74 +83,56 @@ public byte[] ReadContent()
 
         public uint ReadLong()
         {
-            ClearBits();
-            uint result = NetworkOrderDeserializer.ReadUInt32(_memory.Slice(_memoryOffset).Span);
-            _memoryOffset += 4;
+            uint result = NetworkOrderDeserializer.ReadUInt32(Span);
+            _offset += 4;
             return result;
         }
 
         public ulong ReadLonglong()
         {
-            ClearBits();
-            ulong result = NetworkOrderDeserializer.ReadUInt64(_memory.Slice(_memoryOffset).Span);
-            _memoryOffset += 8;
+            ulong result = NetworkOrderDeserializer.ReadUInt64(Span);
+            _offset += 8;
             return result;
         }
 
         public byte[] ReadLongstr()
         {
-            ClearBits();
-            byte[] result = WireFormatting.ReadLongstr(_memory.Slice(_memoryOffset));
-            _memoryOffset += 4 + result.Length;
+            byte[] result = WireFormatting.ReadLongstr(Span);
+            _offset += 4 + result.Length;
             return result;
         }
 
         public byte ReadOctet()
         {
-            ClearBits();
-            return _memory.Span[_memoryOffset++];
+            return _span[_offset++];
         }
 
         public ushort ReadShort()
         {
-            ClearBits();
-            ushort result = NetworkOrderDeserializer.ReadUInt16(_memory.Slice(_memoryOffset).Span);
-            _memoryOffset += 2;
+            ushort result = NetworkOrderDeserializer.ReadUInt16(Span);
+            _offset += 2;
             return result;
         }
 
         public string ReadShortstr()
         {
-            ClearBits();
-            string result = WireFormatting.ReadShortstr(_memory.Slice(_memoryOffset), out int bytesRead);
-            _memoryOffset += bytesRead;
+            string result = WireFormatting.ReadShortstr(Span, out int bytesRead);
+            _offset += bytesRead;
             return result;
         }
 
         public Dictionary<string, object> ReadTable()
         {
-            ClearBits();
-            Dictionary<string, object> result = WireFormatting.ReadTable(_memory.Slice(_memoryOffset), out int bytesRead);
-            _memoryOffset += bytesRead;
+            Dictionary<string, object> result = WireFormatting.ReadTable(Span, out int bytesRead);
```

Comment:
Similar comment like above if it would make sense to buffer the dictionaries or not